### PR TITLE
11 articles queries

### PR DESCRIPTION
**Pull request by philipwalton123**
Add 'sort_by and 'order' query handling including overriding default order, and error handling for invalid query

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -184,16 +184,9 @@ describe('GET /api/articles/:article_id/comments', () => {
...
-        return supertest(app).get('/api/articles/4/comments')
-        .expect(200)
-        .then(response => {
-            expect(response.body.comments).toEqual([])
-        })
-    });
 })
 
-describe('GET /api/articles', () => {
+describe.only('GET /api/articles', () => {
```
Comment (Augs0) [inline __tests__/app.test.js:189]: You might want to take this out to make sure all your test suite is passing :D

**Review by Augs0 — CHANGES_REQUESTED** [review]
Really great testing and code here, but we are missing a query. We want to be able to filter by topic as well. How could we implement this?

File: package.json [hunk 2]
```diff
@@ -34,7 +35,8 @@
   },
```
Comment (emilyb93) [inline package.json:35]: supertest is a dev dependency

File: __tests__/app.test.js [hunk 3]
```diff
@@ -3,7 +3,7 @@ const supertest = require('supertest')
 const seed = require('../db/seeds/seed')
 const db = require('../db/connection')
 const testData = require('../db/data/test-data/index')
-const { forEach } = require('../db/data/test-data/articles')
+require('jest-sorted')
```
Comment (emilyb93) [inline __tests__/app.test.js:6]: we shouldnt need to require in jest-sorted here, it should be added in the package.json as stated in the docs

Jest >v24
Add jest-sorted to your Jest setupFilesAfterEnv configuration. See for help

For example, add the following to your package.json at the root level. See configuring jest for more info.

"jest": {
  "setupFilesAfterEnv": ["jest-sorted"]
} 

add this at root level, remove the require and it should still work

File: models/articles.models.js [hunk 4]
```diff
@@ -51,8 +51,54 @@ exports.getTheseComments = (id) => {
...
+
+    } else {
+        queryStr += ' ORDER BY articles.created_at DESC'
+    }
+
+
+    //handles 'order' query
+    const orderGreenList = ['ASC', 'DESC']
+    
+    if(query.order) {
```
Comment (emilyb93) [inline models/articles.models.js:87]: This block feels like unnecessary and not very dry code. We shouldnt be creating a string, only to later edit it. We should be building the string up. If you have this here, it means you are doing it in the wrong order. Try to refactor so we can do things in the right order without having to use .replace on the string. We shouldnt have to replace anything on a string that you are creating right here and now, and have full control over

**Review by emilyb93 — APPROVED** [review]
Looks good!

**Pull request merged**